modules/cxf/src/main/java/org/mule/module/cxf/MuleInvoker.java

```diff
@@ -125,11 +126,18 @@ else if(e instanceof ComponentException) {
             responseEvent = null;
         }
 
+        if (responseEvent instanceof NonBlockingVoidMuleEvent)
+        {
+            exchange.put(CxfConstants.MULE_EVENT, responseEvent);
+            // Just return something not null to continue the non blocking process
+            return new Object();
```

Comment:
See you added comment, only remaining confusion is that we are retuning null not new Object() elsewhere. line 179.